infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -953,6 +955,20 @@ public void append(List<LogData> range) {
         SegmentHandle firstSh = getSegmentHandleForAddress(first.getGlobalAddress());
         SegmentHandle lastSh = getSegmentHandleForAddress(last.getGlobalAddress());
 
+        // Extract all addresses associated with the provided write range.
+        Set<Long> pendingWrite = range.stream()
+                .map(entry -> entry.getGlobalAddress()).collect(Collectors.toSet());
+
```

Comment:
This is converting a list to a set, a duplicate within the same list won't be detected. Its a non-sorted set anyways, why not just do the overlap check on the List ?

Reply:
I will add a check for duplicates.

There are two reasons for converting it to a set:
1) Relying on the input to be sorted is fragile. If that assumption is ever violated/changed, then this piece of the code will not work correctly.
2) It is conceptually easier to understand/more readable.

Reply:
I'm not saying we should rely on the List to be sorted, I was just pointing out that a linear scan won't be more expensive because the current set intersection logic is as expensive.

Reply:
Done.